core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -0,0 +1,410 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// this function works exactly the same as UIUtils.formatDuration
+function formatDuration(milliseconds, type) {
+  if(type !== 'display') return milliseconds;
+  if (milliseconds < 100) {
+    return milliseconds + " ms";
+  }
+  var seconds = milliseconds * 1.0 / 1000;
+  if (seconds < 1) {
+    return seconds.toFixed(1) + " s";
+  }
+  if (seconds < 60) {
+    return seconds.toFixed(0) + " s";
+  }
+  var minutes = seconds / 60;
+  if (minutes < 10) {
+    return minutes.toFixed(1) + " min";
+  } else if (minutes < 60) {
+    return minutes.toFixed(0) + " min";
+  }
+  var hours = minutes / 60;
+  return hours.toFixed(1) + " h";
+}
+
+function formatStatus(status, type) {
+    if(type !== 'display') return status;
+    if(status) {
+        return "Active"
+    } else {
+        return "Dead"
+    }
+}
+
+function formatBytes(bytes,type) {
+    if(type !== 'display') return bytes;
+    if(bytes == 0) return '0 B';
+    var k = 1000;
+    var dm = 3;
+    var sizes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+    var i = Math.floor(Math.log(bytes) / Math.log(k));
+    return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+}
+
+
+jQuery.extend( jQuery.fn.dataTableExt.oSort, {
+    "title-numeric-pre": function ( a ) {
+        var x = a.match(/title="*(-?[0-9\.]+)/)[1];
+        return parseFloat( x );
+    },
+
+    "title-numeric-asc": function ( a, b ) {
+        return ((a < b) ? -1 : ((a > b) ? 1 : 0));
+    },
+
+    "title-numeric-desc": function ( a, b ) {
+        return ((a < b) ? 1 : ((a > b) ? -1 : 0));
+    }
+} );
+
+$(document).ajaxStop($.unblockUI);
+$(document).ajaxStart(function(){
+    $.blockUI({ message: '<h3>Loading Executors Page...</h3>'});
+});
+
+function createTemplateURI() {
+    var parser = document.createElement('a');
+    var words = parser.baseURI.split('/');
+    var ind = words.indexOf("proxy");
+    if(ind > 0) {
+        var appId = words[ind + 1];
+        var baseURI = words.slice(0, ind + 1).join('/') + '/' + appId + '/static/executorspage-template.html';
+        return baseURI;
+    } else {
+        ind = words.indexOf("history");
+        var baseURI = words.slice(0, ind).join('/') + '/static/executorspage-template.html';
+        return baseURI;
+    }
+}
+
+function createRESTEndPoint() {
+    var parser = document.createElement('a');
+    var words = parser.baseURI.split('/');
```

Comment:
In Safari baseURI is null, it seems to be how Safari handles document.createElement, you may want to research it and possibly find another way to get the URL, there are no other uses of baseURI in the Spark code base.